tool/tsh/common/tsh.go

```diff
@@ -1815,6 +1815,11 @@ func onLogin(cf *CLIConf) error {
 		}
 	}
 
+	// Provide a warning when the auth connector is SSO and the username is specified.
+	if cf.AuthConnector != "" && cf.AuthConnector != constants.LocalConnector && cf.Username != "" {
+		fmt.Fprintf(os.Stderr, "\nwarning: Specifying the Teleport user (%v) is ignored for Single Sign-On (SSO) logins.\nThe SSO authentication will set the user name.\n\n", cf.Username)
```

Comment:
```suggestion
		fmt.Fprintf(os.Stderr, "WARNING: Ignoring Teleport user (%v) for Single Sign-On (SSO) login.\nProvide the user name during the SSO flow instead.\n", cf.Username)
```

Reply:
thanks @zmb3. I have applied this and a recommendation on the `--auth=local` which matches to the gh issue ask.  Please confirm.